path_oidc_test.go

```diff
@@ -122,6 +122,53 @@ func TestOIDC_AuthURL(t *testing.T) {
 		}
 	})
 
+	t.Run("case insensitive", func(t *testing.T) {
+		t.Parallel()
+
+		// normal cases, both passing the role name explicitly and relying on the default
+		for _, rolename := range []string{"test", ""} {
+			data := map[string]interface{}{
+				"role":         rolename,
+				"redirect_uri": "https://EXAMPLE.com",
+			}
+			req := &logical.Request{
+				Operation: logical.UpdateOperation,
+				Path:      "oidc/auth_url",
+				Storage:   storage,
+				Data:      data,
+			}
+
+			resp, err := b.HandleRequest(context.Background(), req)
+			if err != nil || (resp != nil && resp.IsError()) {
+				t.Fatalf("err:%v resp:%#v\n", err, resp)
+			}
+
+			authURL := resp.Data["auth_url"].(string)
+
+			expected := []string{
+				`client_id=abc`,
+				`https://team-vault\.auth0\.com/authorize`,
+				`scope=openid`,
+				`nonce=n_\w{20}`,
+				`state=st_\w{20}`,
+				`redirect_uri=https%3A%2F%2FEXAMPLE.com`,
+				`response_type=code`,
+				`code_challenge=\w+`,
+				`scope=openid`,
+			}
+
+			for _, test := range expected {
```

Comment:
`test` used a variable name is a bit confusing. I wonder if we could rename it to `testPat` or something like that. Also, is there a specific reason why we couldn't have a single regex pattern for the full URL - like, is order of the params not deterministic?

Reply:
I agree about the name `test`. I will update that! 

This was mostly copied from an existing test case. I tried reworking it to use a single regex pattern but it isn't obvious what match is failing and I couldn't get it to work. This way gives us a better error when the match fails.